stock_release_channel/models/stock_picking.py

```diff
@@ -106,18 +106,43 @@ def _find_release_channel_possible_candidate(self):
             .sorted(key=lambda r: (not bool(r.partner_ids), r.sequence))
         )
 
-    def _get_release_channel_possible_candidate_domain(self):
-        self.ensure_one()
+    def _get_release_channel_possible_candidate_domain_channel(self):
         return [
             ("is_manual_assignment", "=", False),
             ("state", "!=", "asleep"),
+        ]
+
+    def _get_release_channel_possible_candidate_domain_picking(self):
+        return [
             "|",
             ("picking_type_ids", "=", False),
             ("picking_type_ids", "in", self.picking_type_id.ids),
             "|",
             ("warehouse_id", "=", False),
             ("warehouse_id", "=", self.picking_type_id.warehouse_id.id),
+        ]
+
+    def _get_release_channel_possible_candidate_domain_partners(self):
+        return [
             "|",
             ("partner_ids", "=", False),
             ("partner_ids", "in", self.partner_id.ids),
         ]
+
+    def _inject_possible_candidate_domain_partners(self):
+        """Hooks that could be overridden.
+
+        Return a boolean.
+        """
+        return True
+
+    def _get_release_channel_possible_candidate_domain(self):
+        self.ensure_one()
+        channel_domain = self._get_release_channel_possible_candidate_domain_channel()
+        picking_domain = self._get_release_channel_possible_candidate_domain_picking()
+        domain = channel_domain + picking_domain
+        if self._inject_possible_candidate_domain_partners():
+            domain.extend(
+                self._get_release_channel_possible_candidate_domain_partners()
+            )
```

Comment:
same here
```suggestion
            expression.AND([
                domain,
                self._get_release_channel_possible_candidate_domain_partners()
            ])
```
or build domains and return `expression.AND(domains)`